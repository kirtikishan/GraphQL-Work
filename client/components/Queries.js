import gql from 'graphql-tag';


export const homePageQuery = gql`
  {
     carOfTheWeek{
          modelId,
          review,
          carModel {
            makeId,
            name,
            price,
            imageUrl
            make {
              name
            }
          }
     }
  }
`;

export const searchQuery = gql`
      query search {
         makes {
            id,
            name
            models {
                id,
                name
            }
         }
      }
    `;


export const detailsQuery = gql`
  query ModelsQuery($id : Int!) {
    model(id: $id) {
      name,
      price,
      imageUrl,
      make {
          id,
          name
      }
    }
  }
`;