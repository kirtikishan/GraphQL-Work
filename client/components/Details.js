import React, {Component} from 'react';
import BEMHelper from 'react-bem-helper';
import gql from 'graphql-tag';
import {graphql} from 'react-apollo';
import {Link} from 'react-router';
import {detailsQuery} from './Queries';

var detailsPage = new BEMHelper('details-page');

export class Details extends Component {
    render() {
        let {model, loading} = this.props.data;

        if (loading) {
            return (<div className="loading"></div>);
        } else {
            return (
                <div {...detailsPage()}>
                    <span {...detailsPage('back')}>
                        <Link to="/search">back</Link>
                    </span>
                    <div {...detailsPage('name-row')}>
                        <span {...detailsPage('make')}>{model.make.name}</span>
                        <span {...detailsPage('name')}>{model.name}</span>
                        <div {...detailsPage('price-box')}>
                            <span {...detailsPage('price-text')}>price</span>
                            <span {...detailsPage('price')}>$ {model.price}</span>
                        </div>
                    </div>
                    <img {...detailsPage('detail-img')} src={model.imageUrl}/>
                </div>
            );
        }
        ;
    }
}

export default graphql(detailsQuery, {
    options: (props) => ({
        variables: {id: props.params.modelId},
    }),
})(Details);

