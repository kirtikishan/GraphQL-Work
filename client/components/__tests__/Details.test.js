import * as React from 'react';
import {Details} from '../Details';
import {shallow} from 'enzyme';
import {detailsQuery} from '../Queries';

describe('HomePage Component', () => {
    it('should render the homepage correctly', () => {
        const data = {
            model: {
                name: "Leaf",
                price: 50000,
                imageUrl: "http://o.aolcdn.com/commerce/autodata/images/USC10NIC161B021001.jpg",
                make: {
                    id: 20,
                    name: "Nissan"
                }
            }
        }

        const wrapper = shallow(
            <Details data={data}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should be rendered the correct home page query', () => {
        expect(detailsQuery).toMatchSnapshot();
    });
})
;
