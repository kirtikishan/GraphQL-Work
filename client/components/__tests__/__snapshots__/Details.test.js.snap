// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component should be rendered the correct home page query 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "ModelsQuery",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "model",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "price",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "imageUrl",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "make",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "id",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 157,
    "start": 0,
  },
}
`;

exports[`HomePage Component should render the homepage correctly 1`] = `
<div
  className="details-page"
>
  <span
    className="details-page__back"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/search"
    >
      back
    </Link>
  </span>
  <div
    className="details-page__name-row"
  >
    <span
      className="details-page__make"
    >
      Nissan
    </span>
    <span
      className="details-page__name"
    >
      Leaf
    </span>
    <div
      className="details-page__price-box"
    >
      <span
        className="details-page__price-text"
      >
        price
      </span>
      <span
        className="details-page__price"
      >
        $ 
        50000
      </span>
    </div>
  </div>
  <img
    className="details-page__detail-img"
    src="http://o.aolcdn.com/commerce/autodata/images/USC10NIC161B021001.jpg"
  />
</div>
`;
